File: fixtures/scale/swift/large/Store/JsonlWriter.swift
import Foundation

struct JsonlWriter {
    let path: String

    func append(_ records: [ResultRecord]) {
        StorePaths.ensureDir()
        let lines = records.map { $0.toLine() }.joined(separator: "\n") + "\n"
        if let handle = FileHandle(forWritingAtPath: path) {
            handle.seekToEndOfFile()
            handle.write(lines.data(using: .utf8)!)
            handle.closeFile()
        } else {
            try? lines.write(toFile: path, atomically: true, encoding: .utf8)
        }
    }
}

File: fixtures/scale/csharp/large/Store/JsonlWriter.cs
using System.Collections.Generic;
using System.IO;

namespace CipherLab.Store
{
    public class JsonlWriter
    {
        private readonly string _path;

        public JsonlWriter(string path)
        {
            _path = path;
        }

        public void Append(IEnumerable<ResultRecord> records)
        {
            StorePaths.EnsureDir();
            using var stream = new StreamWriter(_path, append: true);
            foreach (var record in records)
            {
                stream.WriteLine(record.ToLine());
            }
        }
    }
}
